framework/src/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -408,7 +435,14 @@
             // Parse the rest, with a new receiver.
             FlowExpressionContext newContext = context.changeReceiver(receiver);
             return parse(remainingString, newContext, path, false, true, false,
-                    true, true, false, false, false, true);
+                    true, true, false, false, false, false /*"itself" can only be in the receiver, not in the remaining string*/, true);
```

Comment:
I find the embedded block comment rather tricky. In particular, there is normal code on the far right that's easy to miss on github. I would break the line earlier. Similarly the line comment on line 441.